src/ansys/speos/core/generic/constants.py

```diff
@@ -38,3 +42,347 @@
+    """Bit resolution of create PNG image."""
+
+    png_08 = "png_08"
+    png_10 = "png_10"
+    png_12 = "png_12"
+    png_16 = "png_16"
+
+
+class ColorBalanceModeTypes(str, Enum):
+    """Color Balance Mode types without parameters."""
+
+    none = "none"
+    grey_world = "grey_world"
+
+
+@dataclass
+class BalanceModeUserWhiteParameters:
+    """Parameters for Balance mode: User White."""
+
+    red_gain: float = 1.0
+    """Gain value for Red."""
+    green_gain: float = 1.0
+    """Gain value for green."""
+    blue_gain: float = 1.0
+    """Gain value for blue."""
+
+
+@dataclass
+class BalanceModeDisplayPrimariesParameters:
+    """Parameters for Balance mode: Display primaries."""
+
+    red_display_file_uri: Union[str, Path] = ""
+    """Spectrum path of the red display spectrum."""
+    green_display_file_uri: Union[str, Path] = ""
+    """Spectrum path of the green display spectrum."""
+    blue_display_file_uri: Union[str, Path] = ""
+    """Spectrum path of the red display spectrum."""
+
+
+@dataclass
+class ColorParameters:
+    """Color mode Camera Parameter."""
+
+    balance_mode: Union[
+        ColorBalanceModeTypes.none,
+        ColorBalanceModeTypes.grey_world,
+        BalanceModeUserWhiteParameters,
+        BalanceModeDisplayPrimariesParameters,
+    ] = ColorBalanceModeTypes.none
+    """Camera Balance mode."""
+    red_spectrum_file_uri: str = ""
+    """Path to sensitivity spectrum of red Channel."""
+    green_spectrum_file_uri: str = ""
+    """Path to sensitivity spectrum of green Channel."""
+    blue_spectrum_file_uri: str = ""
+    """Path to sensitivity spectrum of blue Channel."""
+
+
+@dataclass
+class MonoChromaticParameters:
+    """Monochromatic Camera Parameters."""
+
+    sensitivity: str = ""
+    """Path to Sensitivity Spectrum."""
+
+
+@dataclass
+class PhotometricCameraParameters:
+    """Photometric Parameters for Camera."""
+
+    color_mode: Union[MonoChromaticParameters, ColorParameters] = field(
+        default_factory=ColorParameters
+    )
+    """Color mode of the Camera Sensor."""
+    layer_type: Union[LayerTypes.none, LayerTypes.by_source] = LayerTypes.none
+    """Layer separation parameter."""
+    png_bits: PngBits = PngBits.png_16
+    """PNG bit resolution of the Camera Sensor."""
+    wavelength_range: WavelengthsRangeParameters = field(default_factory=WavelengthsRangeParameters)
+    """Wavelength range of the Camera Sensor."""
+    acquisition_integration_time: float = 0.01
+    """Integration Time value for the Camera Sensor."""
+    acquisition_lag_time: float = 0
+    """Acquisition lag time for the Camera Sensor."""
+    gamma_correction: float = 2.2
+    """Gamma correction Value for the Camera Sensor."""
+    transmittance_file_uri: str = ""
+    """Transmittance spectrum location"""
+
+
+@dataclass
+class CameraSensorParameters:
+    """Camera Sensor Parameters."""
+
+    sensor_type_parameters: Union[None, PhotometricCameraParameters] = field(
```

Comment:
same. I dont know if there is a better and cleaner way to do that. best case would be SensorTypes which contains all the types. Need to ask Sebastien